migen/fhdl/structure.py

```diff
@@ -695,6 +695,15 @@ def __getitem__(self, key):
         else:
```

Comment:
Random drive by comment: Should the ClockDomain object maybe implement __eq__?

The code in __getitem__ and __contains__ is very similar..

Feel free to ignore.